src/test/java/core/basesyntax/strategy/impl/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,73 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationHandlerTest {
+    private static final String BANANA = "banana";
+    private static final String APPLE = "apple";
+    private PurchaseOperationHandler purchaseOperationHandler = new PurchaseOperationHandler();
+
+    @BeforeEach
+    void setUp() {
+        Storage.SHOPSTORAGE.clear();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.SHOPSTORAGE.clear();
+    }
+
+    @Test
+    void handle_PurchaseBanana_ok() {
+        Storage.SHOPSTORAGE.put(BANANA, 10);
+        FruitTransaction fruitPurchase = new FruitTransaction();
+        fruitPurchase.setFruit(BANANA);
+        fruitPurchase.setQuantity(5);
+        purchaseOperationHandler.handle(fruitPurchase);
+        assertEquals(5, Storage.SHOPSTORAGE.get(BANANA));
+    }
+
+    @Test
+    void handle_notFruitInStorage_notOk() {
+        FruitTransaction noSuchFruit = new FruitTransaction();
+        noSuchFruit.setFruit(APPLE);
+        noSuchFruit.setQuantity(5);
+        assertThrows(RuntimeException.class,
+                () -> purchaseOperationHandler.handle(noSuchFruit),
+                "Your do not have such fruit");
+    }
+
+    @Test
+    void handle_purchaseFruitTransactionNull_notOk() {
+        assertThrows(NullPointerException.class,
+                () -> purchaseOperationHandler.handle(null),
+                "Your FruitTransaction is Null");
+    }
+
+    @Test
+    void handle_purchaseZeroQuantity_notOk() {

```

Comment:
Why is it not ok? As far as I see your test, don't expect any exceptions, so this flow seems to be ok